src/particle/Particle.h

```diff
@@ -228,6 +260,59 @@ class Particle {
      * @param typeArg the type of this particle
      */
     void setType(int typeArg);
+
+    /**
+     * @brief returns true if this particle is fixed and therefore does not change its position
+     */
+    bool isFixed();
+
+    /**
+     * @brief specify whether this particle should be fixed or not
+     * @param fixedArg true : fixed, false : not fixed
+     */
+    void setFixed(bool fixedArg);
+
+
+    /**
+     * @brief adds the index of a particle as a direct neighbor to this particle 
+     * @param particleIdx the index of the particle to add as a direct neighbor
+     */
+    void addNeighborIdx(int particleIdx);
+
+    /**
+     * @brief adds the index of a particle as a diagonal neighbor to this particle 
+     * @param particleIdx the index of the particle to add as a diagonal neighbor
+     */
+    void addDiagNeighborIdx(int particleIdx);
+
+    /**
+     * @brief returns the index of a direct neighbor particle
+     * @param pos the relative position of the neighbor particle
+     */
+    int getNeighborIdx(DirectNeighborPos pos);
+
+    /**
+     * @brief returns the index of a diagonal neighbor particle
+     * @param pos the relative position of the neighbor particle
+     */
+    int getDiagNeighborIdx(DiagonalNeighborPos pos);
+
+    /**
+     * @brief returns a vector of indices for the direct neighbors of this particle
+     * @retrun a vector of indices for the direct neighbors of this particle
+     */
+    std::vector<int> &getAllDirectNeighborIndices();
+
+    /**
+     * @brief returns a vector of indices for the diagonal neighbors of this particle
+     * @retrun a vector of indices for the direct neighbors of this particle
```

Comment:
```suggestion
     * @return a vector of indices for the direct neighbors of this particle
```
spelling mistake in this is more critical as generated documentation would not be proper. So, be careful with stuff like this, although it's just human error and can happen.

Reply:
yes, thanks!